File: Js/fib.js
/**
 * Devuelve un array con el n‑ésimo término de Fibonacci
 * para cada elemento de entrada.
 *
 * Complejidad: O(m · k) — m = numbers.length, k = mayor n;
 * con k ≤ 30 el tiempo está muy por debajo de 0.5 s.
 */
// Js/fib.js
function fibList(numbers) {
  return numbers.map(n => {
    let a = 0, b = 1;
    for (let i = 0; i < n; i++) [a, b] = [b, a + b];
    return a;
  });
}

if (typeof module !== 'undefined' && module.exports) {
  module.exports = { fibList };
}
export { fibList };

File: python/fib.py
from typing import List

def fib_list(numbers: List[int]) -> List[int]:
    """
    Devuelve el término nésimo de Fibonacci para cada número del array.

    Complejidad: O(m · k)  —m = len(numbers), k = máximo n.
    Con k <= 30 la ejecución está por debajo de 0.5s.
    """
    def fib(n: int) -> int:
        a, b = 0, 1
        for _ in range(n):
            a, b = b, a + b
        return a
    return [fib(n) for n in numbers]
